### ISSUE TITLE
Service Bus: Support for dead_lettering_on_filter_evaluation_exceptions attribute on azurerm_servicebus_subscription

### ISSUE THREAD

**Issue by mjr5749** [hashicorp/terraform-provider-azurerm#8316]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for dead_lettering_on_filter_evaluation_exceptions attribute on azurerm_servicebus_subscription. This was removed in #5718, but I can find no reference in the Azure Service Bus documentation or APIs that this attribute is deprecated, or plans to be deprecated.

This was discussed on the NServiceBus community forum here:
https://discuss.particular.net/t/usage-of-enabledeadletteringonfilterevaluationexceptions-in-asb-transport/2059

SeanFeldman wrote:
> I’ve looked into this and something here doesn’t seem to be right. Subscription’s EnableDeadLetteringOnFilterEvaluationExceptions is not deprecated. You can see it in the current and the future (preview) .NET clients. Moreover, the property deadLetteringOnFilterEvaluationExceptions is still part of the latest version of the ARM template for subscriptions. It’s optional but not deprecated.

### New or Affected Resource(s)

* azurerm_servicebus_subscription

### Potential Terraform Configuration

```hcl
resource "azurerm_servicebus_subscription" "subscription" {
  ...
  dead_lettering_on_filter_evaluation_exceptions = false
  ...
}
```

### References

* Attribute was removed in #5718, may have been deprecated earlier than that.

**Comment by SeanFeldman** [comment]
If the default for ARM template optional parameter `deadLetteringOnFilterEvaluationExceptions` is true, arguably a proper fix would be to fix the default ARM template. I would suggest also raising an issue with the service (broker) team and cross-link the issue to this one. Azure Service Bus broker issue tracker: https://github.com/Azure/azure-service-bus

**Comment by mrichards3** [comment]
@SeanFeldman Our MS support team confirmed that this property is not deprecated, and provided the following explanation of how this property behaves:

> Lets say I have a filter that is:
> x < 1/y
> The subscription will look for messages that have ‘x’ and ‘y’ in it and accept the message, but then, if it finds out that ‘y’ is zero, it's going to throw an exception.
> It will then put the message in the dead letter queue.

This seems like reasonable default behavior, so I'm not sure that the default value of "true" for this property should be changed on the Azure Service Bus end.

**Comment by SeanFeldman** [comment]
My comment was made under the incorrect context, thinking of `EnableFilteringMessagesBeforePublishing`. Please ignore it.
For `EnableDeadLetteringOnFilterEvaluationExceptions` it makes sense to set the default to true.

**Comment by neil-yechenwei** [comment]
Thanks for opening this issue. I assume it's better to get official confirmation from service team. So I raised an issue on Azure/azure-rest-api-specs#10857.

**Comment by mrichards3** [comment]
It appears that this has been resolved by #8402

**Comment by rcskosir** [comment]
I am going to mark this issue as complete with #8402, if that is incorrect, please reach out and I can reopen.
